crates/red_knot_workspace/resources/test/corpus/88_regression_tuple_type_short_circuit.py

```diff
@@ -0,0 +1,17 @@
+"""
+Regression test that makes sure we do not short-circuit here after
+determining that the overall type will be `Never` and still infer
+a type for the second tuple element `2`.
+
+Relevant discussion:
+https://github.com/astral-sh/ruff/pull/15218#discussion_r1900811073
+"""
+
+from typing_extensions import Never
+
+
+def never() -> Never:
+    return never()
```

Comment:
For a more realistic function that would be reasonably annotated as returning `Never`, you could do

```suggestion
    while True:
        pass
```

But it might be more concise to create the tuple in a function, where the function has a parameter annotated with `Never`

Reply:
> For a more realistic function that would be reasonably annotated as returning `Never`

What's wrong with my function? I believe it is correctly typed?

> it might be more concise to create the tuple in a function, where the function has a parameter annotated with `Never`

I didn't do that on purpose. A function that accepts `Never` can not be called (as there is no way to produce an element of type `Never`). So we might later detect its body to be unreachable code?

Reply:
> > For a more realistic function that would be reasonably annotated as returning `Never`
> 
> What's wrong with my function? I believe it is correctly typed?

There's nothing wrong with it at all from a typing perspective! But at runtime, it would obviously cause infinite recursion and not be particularly useful 😆 the more common situation in which a function is annotated as returning `Never` or `NoReturn` in my experience is when a function unconditionally raises an exception, or has an infinite loop.

So this is a very minor nitpick, and I certainly don't object if you ignore it and merge your PR as it is! I just think that as a general principle, it's nice if our test snippets emulate realistic Python code as much as possible :-)

Reply:
I'm afraid I don't really understand why a function returning `Never` with an infinite (no op) loop is more realistic / useful than one with infinite recursion?

You mean if someone *actually* wants to run this, they would prefer an infinite loop because the recursive version would run into "max recursion depth exceeded"? Okay :smile: 

(Note that this is just a corpus test, not a Markdown test)

> I certainly don't object if you ignore it and merge your PR as it is

:+1: